src/tribler/ui/src/dialogs/SaveAs.tsx

```diff
@@ -155,8 +155,11 @@ export default function SaveAs(props: SaveAsProps & JSX.IntrinsicAttributes & Di
 
             if (response === undefined) {
                 setError(`${t("ToastErrorGetMetainfo")} ${t("ToastErrorGenNetworkErr")}`);
+            } else if ('error' in response && typeof response.error === 'object') {
+                let errorCode = response.error as {handled: boolean, code: string, message: string};
+                setError(`${t("ToastErrorGetMetainfo")} ${errorCode.code}`);
             } else if (isErrorDict(response)) {
-                setError(`t("ToastErrorGetMetainfo")} ${response.error}`);
+                setError(`${t("ToastErrorGetMetainfo")} ${response.error}`);
```

Comment:
This is a bit icky, but for the scope of this PR I decided not to make our custom REST API errors consistent with the unhandled errors.